Solopov/DZ5_AjaxAsynchronousMethodAndCSS/ServletTestList/src/main/java/hibernate/dao/DAOEntity.java

```diff
@@ -0,0 +1,89 @@
+package hibernate.dao;
+
+import hibernate.entity.Answer;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.springframework.stereotype.Repository;
+import ui.Question.QuestionAddServlet;
+
+import java.io.Serializable;
+import java.lang.reflect.ParameterizedType;
+import java.util.List;
+
+public abstract class DAOEntity<T, Id extends Serializable>{ //When: T is type of Entity and Id- type of ID
```

Comment:
Where you find this class?)

Reply:
it's my realisation of abstract class. The interface was not an option since the implementations repeat the functional (only the type changes).
Something is wrong?